Jenkinsfile

```diff
@@ -0,0 +1,40 @@
+#!groovy
+
+def nodeLabel = 'docker'
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+node(nodeLabel) {
+  stage 'Checkout'
+  checkout scm
+
+  stage 'Build'
+
+  /* Call the maven build. */
+  mvn "clean install -B"
+
+  /* Archive the test results */
+  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+
+  /* Archive the build artifacts */
+  step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi'])
+}
+
+/* Run maven from tool "mvn" */
+void mvn(def args) {
+    /* Get jdk tool. */
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
```

Comment:
And it is already setting the target bytecode compatibility.  I'm just conservative that I like building with JDK 7 so that I'm not relying on the JDK 8 compiler to generate JDK 7 bytecode.